src/Services/Kros.Authorization.Api/Application/Services/AuthorizationService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using IdentityModel.Client;
+using Kros.AspNetCore.Authorization;
+using Kros.Authorization.Api.Application.Queries;
+using Kros.Utils;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using static IdentityModel.OidcConstants;
+
+namespace Kros.Authorization.Api.Application.Services
+{
+    /// <summary>
+    /// Service for user authorization.
+    /// </summary>
+    public class AuthorizationService : IAuthorizationService
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ApiJwtAuthorizationOptions _apiJwtAuthorizationOptions;
+        private readonly JwtAuthorizationOptions _jwtAuthorizationOptions;
+        private readonly IUserService _userService;
+        private readonly HttpContext _httpContext;
+
+        /// <summary>
+        /// Ctor.
+        /// </summary>
+        /// <param name="httpClientFactory">Http client factory (without name).</param>
+        /// <param name="jwtAuthorizationOptions">Authorization options for authorization service.</param>
+        /// <param name="apiJwtAuthorizationOptions">Authorization options for api service.</param>
+        /// <param name="userService">User service.</param>
+        /// <param name="httpContextAccessor">Httpc context accessor</param>
+        public AuthorizationService(
+            IHttpClientFactory httpClientFactory,
+            IOptions<JwtAuthorizationOptions> jwtAuthorizationOptions,
+            IOptions<ApiJwtAuthorizationOptions> apiJwtAuthorizationOptions,
+            IUserService userService,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            _httpClientFactory = Check.NotNull(httpClientFactory, nameof(httpClientFactory));
+            _apiJwtAuthorizationOptions = Check.NotNull(apiJwtAuthorizationOptions, nameof(apiJwtAuthorizationOptions)).Value;
+            _jwtAuthorizationOptions = Check.NotNull(jwtAuthorizationOptions, nameof(jwtAuthorizationOptions)).Value;
+            _userService = Check.NotNull(userService, nameof(userService));
+            _httpContext = Check.NotNull(httpContextAccessor, nameof(httpContextAccessor)).HttpContext;
```

Comment:
By my opinion is better hold `IHttpContextAccessor` instead of `HttpContext` and in place where we need can use `_httpContextAccessor.HttpContext`.